stl/inc/vector

```diff
@@ -1668,56 +1664,37 @@ private:
         auto& _Right_data = _Right._Mypair._Myval2;
-        if constexpr (is_same_v<_Choose_pocma<_Alty>, _Propagate_allocators>) {
-            _Tidy();
-#if _ITERATOR_DEBUG_LEVEL != 0
-            // intentionally slams into noexcept on OOM, TRANSITION, VSO-466800
-            _My_data._Reload_proxy(_GET_PROXY_ALLOCATOR(_Alty, _Al), _GET_PROXY_ALLOCATOR(_Alty, _Right._Getal()));
-#endif // _ITERATOR_DEBUG_LEVEL != 0
-            _Pocma(_Al, _Right._Getal());
-            _My_data._Take_contents(_Right_data);
-        } else {
-            const pointer _First = _Right_data._Myfirst;
-            const pointer _Last  = _Right_data._Mylast;
-            const auto _Newsize  = static_cast<size_type>(_Last - _First);
-
-            pointer& _Myfirst = _My_data._Myfirst;
-            pointer& _Mylast  = _My_data._Mylast;
 
-            _My_data._Orphan_all();
+        const pointer _First = _Right_data._Myfirst;
+        const pointer _Last  = _Right_data._Mylast;
+        const auto _Newsize  = static_cast<size_type>(_Last - _First);
 
-            const auto _Oldcapacity = static_cast<size_type>(_My_data._Myend - _Myfirst);
-            if constexpr (conjunction_v<bool_constant<_Ptr_copy_cat<_Ty*, _Ty*>::_Trivially_copyable>,
-                              _Uses_default_construct<_Alty, _Ty*, _Ty>, _Uses_default_destroy<_Alty, _Ty*>>) {
-                if (_Newsize > _Oldcapacity) {
-                    _Clear_and_reserve_geometric(_Newsize);
-                }
-
-                _Mylast = _Refancy<pointer>(_Copy_memmove(_Unfancy(_First), _Unfancy(_Last), _Unfancy(_Myfirst)));
-            } else {
-                auto _Oldsize = static_cast<size_type>(_Mylast - _Myfirst);
+        pointer& _Myfirst = _My_data._Myfirst;
+        pointer& _Mylast  = _My_data._Mylast;
 
-                if (_Newsize > _Oldsize) {
-                    if (_Newsize > _Oldcapacity) { // reallocate
-                        _Clear_and_reserve_geometric(_Newsize);
-                        _Oldsize = 0;
-                    }
+        _My_data._Orphan_all();
+        _My_data._Orphan_all();
+        const auto _Oldcapacity = static_cast<size_type>(_My_data._Myend - _Myfirst);
+        if (_Newsize > _Oldcapacity) {
+            _Clear_and_reserve_geometric(_Newsize);
+            _Mylast = _Uninitialized_move(_First, _Last, _My_data._Myfirst, _Al);
+            return;
+        }
 
-                    const pointer _Mid = _First + _Oldsize;
-                    _Move_unchecked(_First, _Mid, _Myfirst);
-                    _Mylast = _Uninitialized_move(_Mid, _Last, _Mylast, _Al);
-                } else {
-                    const pointer _Newlast = _Myfirst + _Newsize;
-                    _Move_unchecked(_First, _Last, _Myfirst);
-                    _Destroy_range(_Newlast, _Mylast, _Al);
-                    _Mylast = _Newlast;
-                }
-            }
+        const auto _Oldsize = static_cast<size_type>(_Mylast - _Myfirst);

```

Comment:
Ditto "do we really need to remove the `memmove` optimization or can we refactor along the lines I suggested above"?

Reply:
Ditto not removed but inside the helper